src/components/navigation.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+
+async function handleShortLinkClick() {
+  const url = 'https://api-ssl.bitly.com/v3/shorten?';
+  // To test this locally either edit your local hosts file or set params to something like the commented line below.
+  //const params = `longUrl=${encodeURIComponent('https://www.mozilla.com')}&access_token=48ecf90304d70f30729abe82dfea1dd8a11c4584&format=json`;
+  const params = `longUrl=${encodeURIComponent(window.location.href)}&access_token=48ecf90304d70f30729abe82dfea1dd8a11c4584&format=json`;
+
+  const response = await fetch(url + params);
+  if (response.status !== 200) {
+    console.error(`Bitly API response error. Status Code: ${response.status}`);
+    return;
+  }
+  const { data } = await response.json();
```

Comment:
Some ES6 fun:

```javascript
const { data: { url } } = await response.json();
```

... if you only care about the URL.

Reply:
Ugly....but I like it.